MaxText/configs/base.yml

```diff
@@ -206,6 +206,10 @@ load_from_prefill_dir: False # If true, decode.py doesn't "prefill" but just rea
 prefill_cache_dir: "" # If set and load_from_prefill_dir, decode.py reads from directory. If set, decode.py writes to directory
 autoregressive_decode_assert: ""
 
+# For nsys profiler, pass the training command to nsys command
+# e.g. nsys profile -s none --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop {training command} 
+profiler: "" # Supported profiler: '', xprof, nsys
```

Comment:
Shall we just set "xprof" as default value here? So that we don't need to override it in `Profiler.__init__()`.

Reply:
This was for backward compatibility. `Profiler.__init__()` is setting the mode to `xprof` only when `enable_profiler` is true but the profiler is not specified. So, by default, the command with `enable_profiler=true` will run xprof profiler and the one with `enable_profiler=false` will not run any profiler.

Reply:
Could we remove the enable_profiler and have just profiler with the three values None | xplane | nsys?

I realize this will break backward compat but we don't promise backward compat and this would be the ideal state moving forward - no need to have two config options that control the same thing. However you would want to post in the MaxText beforehand checking where this needs to changed - there may be some e2e tests outside of maxtext (in xlml) that set enable_profiler